packages/common/src/content/HubContent.ts

```diff
@@ -1,90 +1,85 @@
-import { IItem } from "@esri/arcgis-rest-types";
-import { isMaster } from "cluster";
-import { fetchItemEnrichments } from "../items/_enrichments";
-import { getProp } from "../objects";
-import { getItemThumbnailUrl } from "../resources";
-import { IHubSearchResult } from "../search";
-import { parseInclude } from "../search/_internal/parseInclude";
-import { IHubRequestOptions } from "../types";
-import { getItemHomeUrl } from "../urls";
-import { unique } from "../util";
-import { mapBy } from "../utils";
-import { getFamily } from "./get-family";
-import { getHubRelativeUrl } from "./_internal";
+import { IArcGISContext } from "../ArcGISContext";
+import { HubItemEntity } from "../core/HubItemEntity";
+import { IHubEditableContent, IWithStoreBehavior } from "../core";
 
-/**
- * Enrich a generic search result
- * @param item
- * @param includes
- * @param requestOptions
- * @returns
- */
-export async function enrichContentSearchResult(
-  item: IItem,
-  include: string[],
-  requestOptions: IHubRequestOptions
-): Promise<IHubSearchResult> {
-  // Create the basic structure
-  const result: IHubSearchResult = {
-    access: item.access,
-    id: item.id,
-    type: item.type,
-    name: item.title,
-    owner: item.owner,
-    tags: item.tags,
-    typeKeywords: item.typeKeywords,
-    categories: item.categories,
-    summary: item.snippet || item.description,
-    createdDate: new Date(item.created),
-    createdDateSource: "item.created",
-    updatedDate: new Date(item.modified),
-    updatedDateSource: "item.modified",
-    family: getFamily(item.type),
-    links: {
-      self: "not-implemented",
-      siteRelative: "not-implemented",
-      thumbnail: "not-implemented",
-    },
-  };
+export class HubContent
+  extends HubItemEntity<IHubEditableContent>
+  implements IWithStoreBehavior<IHubEditableContent>
```

Comment:
maybe this isn't necessary or relevant for content right now, but should this also implement `IWithCatalogBehavior` and `IWithSharingBehavior`?

Reply:
content doesn't have a catalog, but yes, it should implement `IWithSharingBehavior`